store/tikv/coprocessor.go

```diff
@@ -173,7 +173,12 @@ func (r *copRanges) slice(from, to int) *copRanges {
 	if to <= len(r.mid) {
 		ran.mid = r.mid[from:to]
 	} else {
-		ran.mid, ran.last = r.mid[from:], r.last
+		if from <= len(r.mid) {
```

Comment:
Yes, the new test iterates all possible combinations of from and to and make sure its behaviour is same as raw slice.